### ISSUE TITLE
More Fortran packages

### ISSUE THREAD

**Issue by ivan-pi** [fortran-lang/fortran-lang.org#232]
I've found a few more Fortran packages I will try to add one by one:

* [ ] Fortran processing library: http://www.biosfera.dea.ufv.br/fpl/index.html
* [ ] 2DECOMP & FFT: http://www.2decomp.org/decomp_api.html
* [ ] SLICOT: http://slicot.org/
* [ ] FORTRAN Routines for Spectral methods: https://cdm.unimore.it/home/matematica/funaro.daniele/rout.htm
* [ ] NUFFT: https://cims.nyu.edu/cmcl/nufft/nufft.html.bak
* [ ] FFTLog: https://jila.colorado.edu/~ajsh/FFTLog/
* [ ] P3DFFT: https://www.p3dfft.net/
* [ ] Parallel FFT Package: https://cs.sandia.gov/~sjplimp/docs/fft/README.html
* [ ] fftMPI: https://fftmpi.sandia.gov/
* [ ] CSLib: https://github.com/lammps/cslib
* [ ] Quadratures for elastostatic Green's functions on flat triangles: https://cims.nyu.edu/cmcl/quadratures/quadratures.html
* [ ] Quadratures for smooth functions in two and three dimensions: https://cims.nyu.edu/cmcl/quadratures/quadratures.html 
* [ ] FMMLIB2D: https://cims.nyu.edu/cmcl/fmm2dlib/fmm2dlib.html 
* [ ] OpenFFT: http://www.openmx-square.org/openfft/

**Comment by ivan-pi** [comment]
My current plan was only to put them in the package index. 👍

**Comment by vmagnin** [comment]
Maybe some yet unlisted packages could be found in the Curlie directory (the historical DMOZ reborn), especially here:
- https://www.curlie.org/en/Computers/Programming/Languages/Fortran/Libraries
- https://www.curlie.org/en/Computers/Programming/Languages/Fortran/Source_Code

**Comment by ivan-pi** [comment]
Indeed, there are a few interesting ones.

On my PC I found the manual for AxiSEM - a parallel spectal-element solver for 3D wave propagation (http://seis.earth.ox.ac.uk/axisem/).

**Comment by ivan-pi** [comment]
This other seismological code (https://github.com/SeisSol/SeisSol) also contains a fair share of Fortran code.

**Comment by ivan-pi** [comment]
On top of the already long list, I've found pencil-code: https://github.com/pencil-code/pencil-code

**Comment by certik** [comment]
There is also Elk (I communicated with the author long time ago):

* [ ] https://elk.sourceforge.io/

**Comment by awvwgk** [comment]
I recommend to send a patch rather than posting the links here, because this makes it easier to discuss the proposed project and allows to actually add them to the package index. Also, it helps to avoid piling up collections of links like the “List of popular Fortran projects,” which are difficult to work on because the relevant context gets lost over time.

**Comment by ivan-pi** [comment]
@krenzland, @ravil-mobile - Given that SeisSol is on track to remove Fortran, I am guessing you don't want SeisSol to be part of the Fortran Package Index (https://fortran-lang.org/packages/)?

**Comment by krenzland** [comment]
Yes, SeisSol isn't really a Fortran code. Nearly everything is written in C++/generated code, some "physics" is still in Fortran. Resulting from this, we have a lot of glue code (which inflates the line count a lot). We probably have around 2k lines of Fortran code that's really relevant.

And no, there isn't any relationship between SeisSol and AxiSEM.

**Comment by gklimowicz** [comment]
Out of curiosity, is anyone creating a Git project with as many of these projects as possible as submodules? I will find myself in need of as large a corpus of Fortran codes as I can lay my hands on, and these are all silos of grain for my mill.

**Comment by ivan-pi** [comment]
Hi @gklimowicz, I'm not aware of any of us within Fortran-Lang having done something like that. Is it compiler testing you have in mind?

Perhaps worth asking @jeffhammond. He was recently searching for Fortran codes [on Twitter](https://twitter.com/jeff_replies/status/1505869184223395842?ref_src=twsrc%5Etfw%7Ctwcamp%5Eembeddedtimeline%7Ctwterm%5Eprofile%3Afortranlang%7Ctwgr%5EeyJ0ZndfZXhwZXJpbWVudHNfY29va2llX2V4cGlyYXRpb24iOnsiYnVja2V0IjoxMjA5NjAwLCJ2ZXJzaW9uIjpudWxsfX0%3D%7Ctwcon%5Etimelinechrome&ref_url=https%3A%2F%2Ffortran-lang.org%2F).

**Comment by jeffhammond** [comment]
https://github.com/Beliavsky/Fortran-code-on-GitHub was useful to me. I wrote a script to clone every repo it references. Was about 25MLOC I think.

I recommend Dalton (GitLab) and NWChem (GitHub) for Fortran testing. Those are both huge codes.  Mostly legacy stuff.

**Comment by gklimowicz** [comment]
@jeffhammond @Beliavsky @ivan-pi Thanks, Jeff and Ivan. I will dig into our chess-playing friend's repo and see if there is anything I've come across to suggest add to it.

I'm planning to work on something of an over-extended Fortran semantic analyzer, so Dalton and NWChem are good, too.

Keep those cards and letters coming. :-)

**Comment by ivan-pi** [comment]
For eigenvalue problems, the package list currently shows only
* ELPA
* ARPACK

We should add the eigenvalue tag to LAPACK (dense eigenvalue problems, successor of EISPACK), SLICOT (generalized eigenvalue problems), PROPACK, and scalapackfx.

A few more libraries for eigenvalues exist, including:

- [ ] FEAST: http://www.feast-solver.org/references.htm (also available in the [NAG library](https://www.nag.com/blog/feast-new-eigensolver-nag-library))
- [ ] SPRAL: https://www.numerical.rl.ac.uk/spral/doc/latest/Fortran/index.html
- [ ] SLEPc: https://slepc.upv.es/ (in C, provides Fortran interface, requires PETSc)
- [ ] PRIMME: https://www.cs.wm.edu/~andreas/software/doc/readme.html (C, provides both F77 and F90 interfaces)
- [ ] EVSL: https://www-users.cse.umn.edu/~saad/software/EVSL/ (C, provides Fortran interface)
- [ ] EigenEXA: https://www.r-ccs.riken.jp/labs/lpnctrt/projects/eigenexa/

Even more single codes/algorithms are mentioned in the following two documents:

* [A Survey of Software for Sparse Eigenvalue Problems | SLEPc](https://slepc.upv.es/documentation/reports/str6.pdf)
* [Software for the Eigenvalue Problem | Sven Hammarling, NAG Ltd & The university of Manchester](http://www.cl.eps.manchester.ac.uk/medialand/maths/archived-events/workshops/www.mims.manchester.ac.uk/events/workshops/MOPNET4/Hammarling.pdf)

**Comment by certik** [comment]
@ivan-pi go ahead and submit a PR with these. Great additions.

**Comment by ivan-pi** [comment]
Hopefully I will find some time soon. 

I just noticed, FEAST has a very nice GUI configurator: http://www.feast-solver.org/config.html. I've been imagining something like this for various Fortran solvers for a long time. For example you could have a GUI for LAPACK, where you select the type of problem, what kind of decomposition you want to use, and it generates the code you can copy-paste into your program.

**Comment by awvwgk** [comment]
Please target the new repository at https://github.com/fortran-lang/webpage for any new submissions for the package index. See https://github.com/fortran-lang/fortran-lang.org/issues/415 for details on the migration.
